internal/pkg/vm/actor/builtin/miner/miner.go

```diff
@@ -299,55 +237,201 @@ var minerExports = exec.Exports{
+	case GetWorker:
+		return reflect.ValueOf((*Impl)(a).GetWorker), signatures[GetWorker], true
+	case GetPeerID:
+		return reflect.ValueOf((*Impl)(a).GetPeerID), signatures[GetPeerID], true
+	case UpdatePeerID:
+		return reflect.ValueOf((*Impl)(a).UpdatePeerID), signatures[UpdatePeerID], true
+	case GetPower:
+		return reflect.ValueOf((*Impl)(a).GetPower), signatures[GetPower], true
+	case AddFaults:
+		return reflect.ValueOf((*Impl)(a).AddFaults), signatures[AddFaults], true
+	case SubmitPoSt:
+		return reflect.ValueOf((*Impl)(a).SubmitPoSt), signatures[SubmitPoSt], true
+	case SlashStorageFault:
+		return reflect.ValueOf((*Impl)(a).SlashStorageFault), signatures[SlashStorageFault], true
+	case ChangeWorker:
+		return reflect.ValueOf((*Impl)(a).ChangeWorker), signatures[ChangeWorker], true
+	case VerifyPieceInclusion:
+		return reflect.ValueOf((*Impl)(a).VerifyPieceInclusion), signatures[VerifyPieceInclusion], true
+	case GetSectorSize:
+		return reflect.ValueOf((*Impl)(a).GetSectorSize), signatures[GetSectorSize], true
+	case GetAsks:
+		return reflect.ValueOf((*Impl)(a).GetAsks), signatures[GetAsks], true
+	case GetAsk:
+		return reflect.ValueOf((*Impl)(a).GetAsk), signatures[GetAsk], true
+	case GetLastUsedSectorID:
+		return reflect.ValueOf((*Impl)(a).GetLastUsedSectorID), signatures[GetLastUsedSectorID], true
+	case GetProvingSetCommitments:
+		return reflect.ValueOf((*Impl)(a).GetProvingSetCommitments), signatures[GetProvingSetCommitments], true
+	case IsBootstrapMiner:
+		return reflect.ValueOf((*Impl)(a).IsBootstrapMiner), signatures[IsBootstrapMiner], true
+	case GetPoStState:
+		return reflect.ValueOf((*Impl)(a).GetPoStState), signatures[GetPoStState], true
+	case GetProvingWindow:
+		return reflect.ValueOf((*Impl)(a).GetProvingWindow), signatures[GetProvingWindow], true
+	case CalculateLateFee:
+		return reflect.ValueOf((*Impl)(a).CalculateLateFee), signatures[CalculateLateFee], true
+	case GetActiveCollateral:
+		return reflect.ValueOf((*Impl)(a).GetActiveCollateral), signatures[GetActiveCollateral], true
+	default:
+		return nil, nil, false
+	}
+}
+
+// InitializeState stores this miner's initial data structure.
+func (*Actor) InitializeState(storage exec.Storage, initializerData interface{}) error {
+	minerState, ok := initializerData.(*State)
+	if !ok {
+		return errors.NewFaultError("Initial state to miner actor is not a miner.State struct")
+	}
+
+	stateBytes, err := encoding.Encode(minerState)
+	if err != nil {
+		return xerrors.Wrap(err, "failed to cbor marshal object")
+	}
+
+	id, err := storage.Put(stateBytes)
+	if err != nil {
+		return err
+	}
+
+	return storage.Commit(id, cid.Undef)
 }
 
 //
-// Exported actor methods
+// vm methods for actor
 //
 
+// Impl is the VM implementation of the actor.
+type Impl Actor
+
+var log = logging.Logger("actor.miner")
+
+// XXX: import cycle, need cleanup
+var XXX_storagemarket_UpdateStorage = types.MethodID(1)
+var XXX_storagemarket_GetProofsMode = types.MethodID(3)
```

Comment:
Please fix anything that can be immediately done so and remove XXX comments before merging. 
- If there's work to be deferred, please write a `TODO` and reference a GitHub issue with the details (for easy grepping).
- If it's just ugly but no immediate action, just leave a comment explaining the situation but without the XXX.

This one is probably worth a discussion offline to choose a path forward.